scripts/import/import_EPMC.pl

```diff
@@ -1282,11 +1242,49 @@ sub check_outdated_citations {
 
     my @split_attrib_id = split /,/, $attrib_id;
 
+    my @outdated_attribs;
+    my @current_attribs;
     foreach my $attrib (@split_attrib_id) {
-      print $wrt "$variation_rsid\t$publication_pmid\t$attrib\n" unless ($citations_pheno_feature->{$attrib.'_'.$publication_pmid} || $citations_pheno_feature_attrib->{$attrib.'_'.$publication_pmid});
+      my $is_selected_attrib_id = $attrib ~~ @attrib_ids;
+
+      my $value = $attrib . '_' . $publication_pmid;
+      my $is_cited_in_pheno = $citations_pheno_feature->{$value} || $citations_pheno_feature_attrib->{$value};
+      if (!$is_selected_attrib_id || $is_cited_in_pheno) {
+        push @current_attribs, $attrib;
+      } else {
+        push @outdated_attribs, $attrib;
+        print $wrt "$variation_rsid\t$publication_pmid\t$attrib\n";
+      }
+    }
+
+    if (@outdated_attribs eq @split_attrib_id) {
+      # remove citation if outdated in all sources
+      $rm_citations_sth->execute($attrib_id, $variation_id, $publication_id) or
+        die "Error: cannot remove $variation_id, $publication_id, $attrib_id from variation_citation\n";
+    } elsif (@outdated_attribs) {
+      # discard outdated sources
+      my $new_attrib_id = join(",", @current_attribs);
+      $update_citations_sth->execute($new_attrib_id, $variation_id, $publication_id) or
+        die "Error: cannot update $variation_id, $publication_id, $attrib_id from variation_citation with new value $new_attrib_id\n";
     }
+  }
 
+  # Remove publications not associated with any variants
+  my $orphan_publications_sth = $dba->dbc()->prepare(qq[
+    select publication_id, title
+    from publication
+    where publication_id NOT IN
+      (select distinct publication_id from variation_citation) ]);
+  $orphan_publications_sth->execute() or die;
+  my $orphan_publications = $orphan_publications_sth->fetchall_arrayref();
+
+  if ($orphan_publications) {
+    print $wrt "\nDeleted publications with no associated variants (publication_id, title):\n";
+    remove_publications($pub_ad, $wrt, $orphan_publications);
+  } else {
+    print $wrt "\nNo publications with no associated variants to remove.\n";
```

Comment:
The evidence_attribs is only being updated by `remove_publications()`. If no publications are removed then the evidence_attribs is never updated, this is a problem when a variant doesn't have any more citations.
An alternative is to update [update_evidence()](https://github.com/Ensembl/ensembl-variation/blob/59efd6ddd9e3c77019f45624852248fec8905cf5/scripts/import/import_EPMC.pl#L595), this method could add/remove cited evidence instead of just adding it.

Reply:
Updated the code: after removing a citation, it will remove the 'Cited' attribute for a variant if the variant is not in `variation_citation` anymore.